### ISSUE TITLE
[BUG] in PUT /v2/entities/<id>/attrs/<attr_name> if try to update a metadata sometimes it is duplicated

### ISSUE THREAD

**Issue by iariasleon** [telefonicaid/fiware-orion#1433]
In `PUT /v2/entities/<id>/attrs/<attr_name>` if try to update a metadata value only, but the metadata created previously has metadata type, that so, the metadata is duplicated but without metadata type.

#### Create request:
```
POST http://localhost:1026/v2/entities
Content-Type: application/json
Fiware-Service: test_attribute_metadata_name_update_4
Fiware-ServicePath: /test
{
      "type": "house", 
      "temperature": {
              "type": "celcius", 
              "value": "34", 
              "house_flat_1": {"type": "alarm", "value": "hot"}, 
              "house_flat_0": {"type": "alarm", "value": "hot"}
       }, 
       "id": "room_7"
}
```

#### Update request:
```
PUT http://localhost:1026/v2/entities/room_7/attrs/temperature
Content-Type: application/json
Fiware-Service: test_attribute_metadata_name_update_4
Fiware-ServicePath: /test
 {
      "house_flat_1": "5678", 
      "house_flat_0": "5678"
}
```

#### doc in mongo:
```
{ 
         "_id" : { "id" : "room_7", "type" : "house", "servicePath" : "/test" }, 
         "attrNames" : [ "temperature" ], 
         "attrs" : { 
               "temperature" : { 
                       "value" : "34", 
                       "type" : "celcius", 
                       "md" : [ 
                               { "name" : "house_flat_1", "value" : "5678" }, 
                               { "name" : "house_flat_0", "value" : "5678" }, 
                               { "name" : "house_flat_1", "type" : "alarm", "value" : "hot" },
                               { "name" : "house_flat_0", "type" : "alarm", "value" : "hot" } ], 
                       "creDate" : 1446032114, 
                       "modDate" : 1446032114 } }, 
         "creDate" : 1446032114, 
         "modDate" : 1446032114 
}
```

**Comment by raconte** [comment]
#### Create entity
```
(curl localhost:1026/v2/entities --header Content-Type:application/json -d @- | python -mjson.tool)<< EOF
{
  "type": "Room",
  "temperature": {
    "value": "celsius",
    "type": "34",
	"house_flat_1": {"type": "alarm", "value": "hot"}, 
    "house_flat_0": {"type": "alarm", "value": "hot"}
  },
  "id": "Room11"
}
EOF
```

#### Returns
```
{
    "description": "unrecognized property for context attribute",
    "error": "BadRequest"
}
```
#### Create entity
```
(curl localhost:1026/v2/entities --header Content-Type:application/json -d @- | python -mjson.tool)<< EOF
{
  "type": "Room",
  "id": "Room10",
  "humidity": {
    "value": 78
  },
  "temperature": {
    "value": "celsius",
    "type": "34",
    "metadata": {
			"house_flat_1": {"type": "alarm", "value": "hot"}, 
            "house_flat_0": {"type": "alarm", "value": "hot"}
    }
  }
}
EOF
```
Created 

#### PUT
``` 
(curl localhost:1026/v2/entities/Room10/attrs/temperature --header Content-Type:application/json -X PUT -d @- | python -mjson.tool )<< EOF 
{
      "house_flat_1": "5678", 
      "house_flat_0": "5678"
}
EOF
```
#### Returns

```
{
    "description": "unrecognized property for context attribute",
    "error": "BadRequest"
}
```

**Comment by fgalan** [comment]
(It has been automatically closed by github. Reopening)

**Comment by fgalan** [comment]
Issue fixed in PR #1740. Please @iariasleon have check if it has been finaly solved.

**Comment by iariasleon** [comment]
LGTM
```
{"id":"room_7","type":"house","temperature":{"type":"celsius","value":"dfgdfgdf","metadata":{"house_flat_1":{"type":null,"value":"5678"},"house_flat_0":{"type":null,"value":"5678"}}}}
```
